File: Starter Problems 3/dict.py
op = open('dictin.txt', 'r')

'''
d is the number of entries in the biliilingsdgl language
w is the number of translations
'''
d, w = map(int, op.readline().split())
dic = {}

for i in range(d):
	a, b = op.readline().split()
	dic[a] = b

ot = open('dictout.txt', 'w')
for i in range(w):
	n = op.readline().strip()
	if n in dic.keys():
		ot.write(dic[n] + '\n')
	else:
		ot.write('C?\n')

File: Starter Problems 3/dict.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ifstream inFile;
    inFile.open("dictin.txt");
    int d,w;
    inFile >> d >> w;
    map<int,int> translations;
    int tempa,tempb;
    for (int i=0;i<d;i++) {
        inFile >> tempa >> tempb;
        translations[tempa] = tempb;
    }
    ofstream outFile;
    outFile.open("dictout.txt");
    int temp;
    for (int i=0;i<w;i++) {
        inFile >> temp;
        //printf("%d\n",temp);
        map<int,int>::iterator it = translations.find(temp);
        if (it != translations.end()) {
            //printf("%d %d\n",it->second,temp);
            outFile << it->second;
        }
        else {
            outFile << "C?";
        }
        outFile << "\n";
    }
    inFile.close();
    outFile.close();
}
